packages/mock-block-dock/src/MockBlockDock.tsx

```diff
@@ -21,41 +23,54 @@ export const MockBlockDock: VoidFunctionComponent<MockBlockDockProps> = ({
   children,
   blockSchema,
 }) => {
-  const blockEntityType: BlockProtocolEntityType = {
-    entityTypeId: "blockType1",
-    title: "BlockType",
-    type: "object",
-    $schema: "https://json-schema.org/draft/2019-09/schema",
-    $id: "http://localhost/blockType1",
-    ...(blockSchema ?? {}),
-  };
+  const getMockData = useCallback(
+    (mockData: MockData): MockData => {
+      const blockEntityType: BlockProtocolEntityType = {
+        entityTypeId: "blockType1",
+        title: "BlockType",
+        type: "object",
+        $schema: "https://json-schema.org/draft/2019-09/schema",
+        $id: "http://localhost/blockType1",
+        ...(blockSchema ?? {}),
+      };
 
-  const initialBlockEntity: BlockProtocolEntity = {
-    accountId: "account1",
-    entityId: "block1",
-  };
+      const initialBlockEntity: BlockProtocolEntity = {
+        accountId: "account1",
+        entityId: "block1",
+      };
 
-  if (
-    children.props &&
-    typeof children.props === "object" &&
-    Object.keys(children.props).length > 0
-  ) {
-    Object.assign(initialBlockEntity, children.props);
-  }
+      if (
+        children.props &&
+        typeof children.props === "object" &&
+        Object.keys(children.props).length > 0
+      ) {
+        Object.assign(initialBlockEntity, children.props);
+      }
 
-  initialBlockEntity.entityTypeId = blockEntityType.entityTypeId;
+      initialBlockEntity.entityTypeId = blockEntityType.entityTypeId;
 
-  mockData.entities.push(initialBlockEntity);
-  mockData.entityTypes.push(blockEntityType);
+      const nextMockData: MockData = { ...mockData };
 
-  const { entities, entityTypes, links, functions } =
-    useMockDatastore(mockData);
+      nextMockData.entities = [...mockData.entities, initialBlockEntity];
+      nextMockData.entityTypes = [...mockData.entityTypes, blockEntityType];
+      nextMockData.blockEntityId = children.props.entityId;
 
-  const latestBlockEntity = useMemo(
-    () =>
-      entities.find((entity) => entity.entityId === children.props.entityId),
-    [entities, children.props.entityId],
+      return nextMockData;
+    },
+    [blockSchema, children.props],
+  );
+
+  const [mockData, setMockData] = useState<MockData>(
```

Comment:
Move `mockData` to state, to make it more stable. The previous code was running on each render, causing multiple versions of the same entity to be pushed into the `entities` array.

Additionally, the computing code has been moved to a callback function above.

Reply:
could `mockData` be set by the return of `useMemo` instead to make it stable? Then you wouldn't need this state or the effect below.